tubemq-server/src/main/java/org/apache/tubemq/server/broker/BrokerServiceServer.java

```diff
@@ -673,7 +673,7 @@ public SendMessageResponseB2P sendMessageP2B(SendMessageRequestP2B request,
             final MessageStore store =
                     this.storeManager.getOrCreateMessageStore(reqTopic, partition);
             final long messageId = this.idWorker.nextId();
-            if (store.appendMsg(messageId, dataLength, checkSum, msgData,
+            if (store.appendMsg(messageId, dataLength, request.getCheckSum(), msgData,
```

Comment:
This place needs to be reserved. Even if the client does not do CRC, it needs to check whether it is consistent with the storage during consumption. After removal, this logic will be lost

Reply:
yes, I see.

Reply:
I will repost laster